### ISSUE TITLE
BUILD FAILURE

### ISSUE THREAD

**Issue by teguheka** [htrc/HTRC-Tools-Wso2BackupRestore#1]
[ERROR] Error while parsing schema(s).Location [ file:/Documents/HTRC-Tools-Wso2BackupRestore/src/main/resources/schemas/backup.xsd{20,64}].
org.xml.sax.SAXParseException; systemId: file:/Documents/HTRC-Tools-Wso2BackupRestore/src/main/resources/schemas/backup.xsd; lineNumber: 20; columnNumber: 64; src-resolve: Cannot resolve the name 'security:Role' to a(n) 'type definition' component.
        at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203)

any idea for this issue?

**Comment by borice** [comment]
Hi.  Thanks for letting me know.  Not sure what happened as it worked fine last time I checked. 
Anyway - I fixed it in dda8c646239d9e5824606e03e52c150fc185c48c

**Comment by teguheka** [comment]
yes success to build, thanks, btw i have another problem.

```
org.hathitrust.htrc.wso2.tools.backuprestore.exceptions.BackupRestoreException: Error retrieving public files

Caused by: org.wso2.carbon.registry.core.exceptions.ResourceNotFoundException: Resource does not exist at path /htrc/_PUBLIC_/files

```
any advice master Borice?

**Comment by teguheka** [comment]
btw this one can use for wso2 apim, is it right?

**Comment by borice** [comment]
This is an internal tool used at HTRC (Hathitrust Research Center) and is not meant to be used for generic WSO2 backup and restore functionality.  This tool makes a number of assumptions that are particular to our use.... so if you want a WSO2 backup and restore tool that works for your purpose, you'll have to modify the code.  There is a lot of code that's already generic, but you'll have to change/remove the parts that do not apply to your setup.

Also, this tool is being used for WSO2 Identity Server.... it would work for other WSO2 products, but it does not back up anything specific to other tools. I would guess that for APIM you'd want to backup some specific things also. 

Bottom line: this tool, as written, won't work for your purpose.  But it wouldn't be terribly hard to modify to make it fit whatever purpose you need.

**Comment by teguheka** [comment]
aah i see, thanks borice
